### PR TITLE
#1524 Allow custom search placeholder in Flexible Table

### LINKED ISSUE #1524: Flexible table enhancements
- Add the ability to pass in jsx for the empty table object instead of only text
- Allow for custom search placeholder. Add a new `searchPlaceholder` prop

### PR DESCRIPTION
**Pull request by nmgokhale**
Closes #1524 

- `emptyTablePlaceholder` prop in Flexible table supports "string" and "element" 
- Added a new prop `searchPlaceholder` (string) to customize search text

Default search text - `"Find in column"`
Screenshot 2023-07-13 at 3 13 23 PM

Customized search text - 
Screenshot 2023-07-13 at 3 13 30 PM

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: canvas_modules/common-canvas/locales/common-properties/locales/en.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
   "properties.support.message": "If the problem persists, contact support.",
   "properties.unexpected.message": "Oops, the pane didn't load.",
   "table.search.header": "View in tables",
-  "table.search.placeholder": "Find in column {column_name}",
```
Comment (matthoward366) [inline canvas_modules/common-canvas/locales/common-properties/locales/en.json:33]: I think we could keep this.  Is there a reason it was changed?

File: canvas_modules/common-canvas/src/common-properties/components/field-picker/field-picker.jsx [hunk 2]
```diff
@@ -418,6 +418,7 @@ export default class FieldPicker extends React.Component {
 				updateRowSelections={this.updateFieldSelections}
 				rowSelection={ROW_SELECTION.MULTIPLE}
 				light={this.props.controller.getLight() && !this.props.controller.isTearsheetContainer()}
+				searchPlaceholder="Custom search placeholder"
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/components/field-picker/field-picker.jsx:421]: was this done for testing?  I don't think we actually want this set.

File: canvas_modules/common-canvas/locales/common-properties/locales/en.json (hunk 1 above)
Comment (nmgokhale) [reply canvas_modules/common-canvas/locales/common-properties/locales/en.json:33]: I thought we didn't want column name in the default search message. I will revert this change.

**Review by matthoward366 — APPROVED** [review]
lgtm

**Pull request merged**